File: app/src/main/java/com/raju/javabaseproject/ui/adapters/delegate/base/AdapterDelegate.java
package com.raju.javabaseproject.ui.adapters.delegate.base;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

/**
 * Created by Rajashekhar Vanahalli on 12/21/17.
 */

public interface AdapterDelegate<T> {

    boolean isForViewType(@NonNull T items, int position);

    @NonNull
    RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent);

    void onBindViewHolder(@NonNull T items, int position, @NonNull RecyclerView.ViewHolder holder);

    void setDelegateClickListener(DelegateClickListener delegateClickListener);
}

File: app/src/main/java/com/raju/javabaseproject/ui/adapters/delegate/base/AdapterDelegate.kt
package com.raju.javabaseproject.ui.adapters.delegate.base

import android.view.ViewGroup
import androidx.annotation.NonNull
import androidx.recyclerview.widget.RecyclerView

interface AdapterDelegate<T> {
    fun isForViewType(@NonNull items: T, position: Int): Boolean
    fun onCreateViewHolder(parent: ViewGroup): RecyclerView.ViewHolder
    fun onBindViewHolder(@NonNull items: T, position: Int, holder: RecyclerView.ViewHolder)
    fun setDelegateClickListener(delegateClickListener: DelegateClickListener)
}